niwidgets/streamlines.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import print_function
+import nibabel as nib
+import numpy as np
+from ipywidgets import interact, fixed, widgets
+import os
+import ipyvolume as ipv
+
+
+def length(x):
+    """Returns the sum of euclidean distances between neighboring points"""
+    return np.sum(np.sqrt(np.sum((x[:-1, :] - x[1:, :]) *
+                                 (x[:-1, :] - x[1:, :]),
+                                 axis=1)))
+
+
+def color(x):
+    """Returns an approximation for color of line based on its endpoints"""
+    dirvec = (x[0, :] - x[-1, :])
+    return (dirvec/(np.sqrt(np.sum(dirvec*dirvec, axis=-1)))).dot(np.eye(3))
+
+
+class StreamlineWidget:
+    """
+    Turns mrtrix track or trackvis files into interactive plots using ipyvolume
+
+    Args
+    ----
+        filename : str
+                The path to your ``.trk`` file. Can be a string, or a
+                ``PosixPath`` from python3's pathlib.
+    """
+
+    def __init__(self, filename):
+
+        if not os.path.isfile(filename):
+            # for Python3 should have FileNotFoundError here
+            raise IOError('file {} not found'.format(filename))
+
+        if not nib.streamlines.is_supported(filename):
+            raise ValueError(('File {0} is not a streamline file supported '
+                              'by nibabel').format(filename))
+
+        self.filename = filename
+
+    def plot(self, skip=10, **kwargs):
+        """
+        This is the main method for this widget.
+
+        Args
+        ----
+            skip : int
```

Comment:
The explanation of this parameter could be understood as the number of streamlines to omit from the beginning of the collection, or something of that sort. 

Might be easier if it is formulated as `proportion_to_display`, from which the skip is calculated. 

BTW: the default here and in the `_default_plotter` method is different. Is that intentional? 

Also, what would a `None` input do here?

Reply:
`None` would use all streamlines.  i will change this to `proportion_to_display`.

the defaults were not updated since the efficiency boost. initially we couldn't display more than 5k lines, but now it can handle much more.